Mathlib/Algebra/Category/ModuleCat/Presheaf.lean

```diff
@@ -416,3 +416,56 @@ lemma restrictionApp_toPresheafOfModules {X Y : Cᵒᵖ} (f : X ⟶ Y) :
     PresheafOfModules.restrictionApp f M.toPresheafOfModules = M.map f := rfl
 
 end BundledCorePresheafOfModules
+
+namespace PresheafOfModules
+
+variable (R)
+
+/-- Auxiliary definition for `unit`. -/
+def unitCore : CorePresheafOfModules R where
+  obj X := R.obj X
+  map {X Y} f := by
+    exact
+      { toFun := (R.map f).toFun
+        map_add' := by simp
+        map_smul' := by simp }
+
+/-- The obvious free presheaf of modules of rank `1`. -/
+abbrev unit : PresheafOfModules R := (unitCore R).toPresheafOfModules
+
+lemma unit_map_one {X Y : Cᵒᵖ} (f : X ⟶ Y) : (unit R).map f (1 : R.obj X) = (1 : R.obj Y) :=
```

Comment:
```suggestion
lemma unit_map_one {X Y : Cᵒᵖ} (f : X ⟶ Y) : (unit R).presheaf.map f (1 : R.obj X) = (1 : R.obj Y) :=
```
It might be a good idea to write the lemma like this. This makes it so that the `erw [unit_map_one]` below would become a `rw`. Then we have the `simp` lemma `PresheafOfModules.map_apply` which could be used to deduce the version without the `.presheaf`

Reply:
I don't like that it makes it longer though. I think `(unit R).map` should be the preferred spelling. For the `erw` problem, maybe we should make `PresheafOfModules.map` reducible, but it might be a head symbol and it might not work. Anyway these problems are everywhere in the AG library and I believe some smart people are thinking about how to solve it (Kim, Johan, Dagur etc.) and I think we can leave this as is for now. 
Hopefully we can have a thorough discussion or brainstorm at the AG workshop.

Reply:
To clarify, I'm not stopping you from figuring this out. Another smart mind gives a greater chance in coming up with a better solution! It's just that I believe it is a bigger problem and I don't think it is possible to solve it here.

Reply:
Okay thanks for the answer! I wasn't aware of how widespread these issues were (and that some people are thinking on how to solve it!), then its fair to leave it as it is I think :)

Reply:
There is a mathematical issue here also: when studying morphisms of presheaves of modules, sometimes we want to look at the underlying morphisms of presheaves of abelian groups (this is `.presheaf`), and sometimes we want to look at the family of linear maps between modules of sections (this is `Hom.app`). I would think that `Hom.app` should be the better spelling when applicable.

Reply:
But then we can at least get rid of `erw [unit_map_one]` by doing `rw [← map_apply, unit_map_one]`

Reply:
> There is a mathematical issue here also: when studying morphisms of presheaves of modules, sometimes we want to look at the underlying morphisms of presheaves of abelian groups (this is related to `.presheaf`), and sometimes we want to look at the family of linear maps between modules of sections (this is `Hom.app`). I would think that `Hom.app` should be the better spelling when applicable.

I agree with this. Do you have any thoughts on the `@[simp]` from `map_apply` in this file then, i.e. should it be removed? Because then `dsimp` would automatically try to phrase things in terms of the underlying presheaves of abelian groups, but maybe thats not a big problem?

Reply:
Thanks for the suggestion! I have slightly improved simp lemmas in this commit https://github.com/leanprover-community/mathlib4/pull/13461/commits/ed087fd3bc36f70c3944968e8ef9d828f10b8382